here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/query/WhereClauseBuilder.kt

```diff
@@ -95,13 +98,17 @@ class WhereClauseBuilder(private val request: ReadFeatures) {
             )
 
             is SpIntersects -> {
-                // TODO: Add transformations!
-                val twkb = PgUtil.encodeGeometry(
-                    spatial.geometry,
-                    Flags().geoGzipOff().geoEncoding(GeoEncoding.TWKB)
-                )
-                val placeholder = placeholderForArg(twkb, PgType.BYTE_ARRAY)
-                where.append("ST_Intersects(naksha_geometry(${PgColumn.geo}, ${PgColumn.flags}), ST_GeomFromTWKB($placeholder))")
+                val geometryToCompare = when(spatial.transformation) {
+                    null -> {
+                        val twkbPlaceholder = placeholderForArg(
+                            encodedGeo(spatial.geometry, TWKB),
+                            PgType.BYTE_ARRAY
+                        )
+                        "ST_GeomFromTWKB($twkbPlaceholder)"
+                    }
+                    else -> resolveTransformation(spatial.transformation!!, spatial.geometry)
+                }
```

Comment:
Can you change this to avoid the compiler-check-bypass, so instead of using `spatial.transformation!!`, copy the member to the stack, and then use from there, which theoretically is the right thing to do, and even makes the code minimally faster:
```kotlin
val geometryToCompare = when(val transform = spatial.transformation) {
    null -> {
        val twkbPlaceholder = placeholderForArg(
            encodedGeo(spatial.geometry, TWKB),
            PgType.BYTE_ARRAY
        )
        "ST_GeomFromTWKB($twkbPlaceholder)"
    }
    else -> resolveTransformation(transform, spatial.geometry)
}
```

Reply:
done with simple cast, as discussed on webex - TWKB will be used everywhere